### PR TITLE
feature(UI)#169998778: user can sign in

### PR DESCRIPTION
**Pull request by mic7x4**
#### What does this PR do?
This PR contains the user signIn endpoint 
#### Description of Task to be completed?
have the following Endpoint
         -POST /api/v1/signin
#### How should this be manually tested?
after cloning this repository cd in directory open terminal run npm start make sure you have postman installed in order to see how it works
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#169998778
#### Questions:

### CODE REVIEW

File: server/controllers/userControllers.js [hunk 1]
```diff
@@ -15,39 +15,63 @@ class userControllers {
...
+      username,
+      isAdmin: false,
+    };
+
     const hashedPassword = hashPassword(password);
     const userToken = signJwt(newUser);
     newUser.password = hashedPassword;
-    if (email === newUser.email) {
+    const userEmail = newUser.email;
+    if (email === userEmail) {
```
Comment (YvesIraguha) [inline server/controllers/userControllers.js]: Hello, @mic7x4 why this condition?

File: server/controllers/userControllers.js [hunk 2]
```diff
@@ -15,39 +15,63 @@ class userControllers {
...
           phoneNumber,
           username,
+          isAdmin: false,
         },
       });
     }
     return res.status(400).json({
       status: res.statusCode,
-      message: 'email doesn\'t match',
+      message: "email doesn't match",
```
Comment (YvesIraguha) [inline server/controllers/userControllers.js]: @mic7x4 email doesn't match with what?

File: server/controllers/userControllers.js [hunk 3]
```diff
@@ -15,39 +15,63 @@ class userControllers {
...
+
+  //   User signin Controllers
+  static userSignin(req, res) {
+    const { email, password } = req.body;
+    const userwithEmail = Users.find((user) => user.email === email);
+    const userEmail = userwithEmail.email;
+    const findPassword = userwithEmail.password;
+    const compare = compareHash(password, findPassword);
+    const userToken = signJwt(email);
+    if (userEmail === email && compare === true) {
```
Comment (YvesIraguha) [inline server/controllers/userControllers.js]: @mic7x4 why this condition?

File: server/routes/userRoutes.js [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+import userController from '../controllers/userControllers';
+import Users from '../model/Users';
+import validation from '../validation/userValidation';
 
 const userRoute = express.Router();
 // get all users
-userRoute.get('/users',(req, res)=>{
-    return res.status(200).json({ users: Users });
-})
+userRoute.get('/users', (req, res) => res.status(200).json({ users: Users }));
```
Comment (YvesIraguha) [inline server/routes/userRoutes.js:8]: @mic7x4 can you think about moving this controller to users' controller file?

File: server/tests/userTests.js [hunk 5]
```diff
@@ -46,5 +53,28 @@ describe('Test All user Endpoints', () => {
...
         });
     });
+    it('It should Sign the user in ', (done) => {
+      chai
+        .request(app)
+        .post('/api/v1/auth/signin')
+        .send(correctLoginUser)
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.should.have.property('status');
```
Comment (YvesIraguha) [inline server/tests/userTests.js:64]: @mic7x4 can you expect/test the keys of the `response.body` and their values as well?

**Review by YvesIraguha — CHANGES_REQUESTED** [review]

File: server/controllers/userControllers.js (hunk 3 above)
Comment (mic7x4) [reply server/controllers/userControllers.js]: I wanna check if the email given and the password user typed is equal to the hashed password

File: server/routes/userRoutes.js (hunk 4 above)
Comment (mic7x4) [reply server/routes/userRoutes.js:8]: yeah

File: server/controllers/userControllers.js (hunk 2 above)
Comment (mic7x4) [reply server/controllers/userControllers.js]: Email given or Entered doesn't match with the one we have in our []

**Pull request merged**